### PR TITLE
Demo Travis CI build 

### PR DESCRIPTION
**Pull request by mugdha-adhav**
## Summary
- Basic Travis CI build for building the Go application server and CLI
- Runs the build on linux and OS X
- Add travis build link in the README

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+language: go
+
+os: 
+  - linux
+  - osx
+
+branches:
+  only: 
+    - master
```
Comment (mbtamuli) [inline .travis.yml]: The default branch is `main` and not `master`

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+sudo: false
```
Comment (mbtamuli) [inline .travis.yml]: _If you currently specify `sudo: false` in your `.travis.yml`, we recommend removing that configuration._

https://blog.travis-ci.com/2018-11-19-required-linux-infrastructure-migration

**Review by mbtamuli — CHANGES_REQUESTED** [review]
Always run your Travis CI configuration through the [Travis CI Build Config Explorer](https://config.travis-ci.com/explore)

I got these warnings/suggestions for your config
```
[warn] on root: deprecated key: "sudo" (The key `sudo` has no effect anymore.)

[info] on root: missing dist, using the default "xenial"
```

File: .travis.yml (hunk 1 above)
Comment (mugdha-adhav) [reply .travis.yml]: oops 😥

File: .travis.yml (hunk 2 above)
Comment (mugdha-adhav) [reply .travis.yml]: I referred it from Dave Cheney's blog: https://dave.cheney.net/2018/07/16/using-go-modules-with-travis-ci
But the blog is pretty old.

File: .travis.yml (hunk 1 above)
Comment (mugdha-adhav) [reply .travis.yml]: Fixed!

**Review by mbtamuli — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**